src/axom/spin/internal/linear_bvh/BVHData.hpp

```diff
@@ -66,11 +66,11 @@ struct BVHData
     m_leaf_nodes( nullptr )
   {}
 
-  void allocate( int32 size )
+  void allocate( int32 size, int allocID )
   {
     AXOM_PERF_MARK_FUNCTION( "BVHData::allocate" );
-    m_inner_nodes = axom::allocate< Vec< FloatType,4 > >( (size-1)*4 );
-    m_leaf_nodes  = axom::allocate< int32 >( size );
+    m_inner_nodes = axom::allocate< Vec< FloatType,4 > >( (size-1)*4, allocID );
```

Comment:
Magic number? A comment here about what '4' refers to would be helpful. Alternatively use a named const variable instead.

Reply:
Yeah, the `Vec< FloatType, 4 >` is the underlying internal data-type used by the BVH. No doubt it is magical. 

I'll try to clean this up some (but, perhaps in a subsequent PR).

Reply:
I think adding a named integer constant and using it everywhere is all that is needed.

Reply:
I'll fix it here. In other parts of the code, I have defined a template alias `vec4_t`, I think doing that in a more global scope inside the BVH internals would help.

- [ ] cleanup

Reply:
Done.